### ISSUE TITLE
How do you handle Transform in Entitas?

### ISSUE THREAD

**Issue by c0ffeeartc** [sschmid/Entitas-CSharp#741]
Hi guys, how do you solve parenting and complex transformations in Entitas + Unity?

For example I need a cube stepping by rotating around it's edge. I'm sending an event `( stepBegnPos, stepEndPos, normalizedProgress )` and view reacts to it. The view can update itself with `Transform.RotateAround` or with parenting cube to pivot and rotating the pivot by angle.
The game is real time arcade and I need to track all transform fields at any given time.
Probably the correct way would be to do transformations in unity and replace position, rotation, scale in entity if the value differs, view becomes controller and I don't like it. 

There are many other places where transforms and parenting are used extensively, and I'd like to have a robust solution instead of hack fixes

**Comment by zhuchun** [comment]
Hi, have you seen this article [How I build games with Entitas](https://github.com/sschmid/Entitas-CSharp/wiki/How-I-build-games-with-Entitas-%28FNGGames%29)? Personally, I take the whole ECS system as a big simulator, so everything inside the core determines how my game behave and the view layer only cares about how data be displayed.

**Comment by c0ffeeartc** [comment]
@zhuchun yes I read this link, and my implementation is very close to it.
 What I lack is some kind of TransformController wrapper or a set of components, that have parenting and matrix transformation functions. How do you deal with transformations under that architecture?

**Comment by c0ffeeartc** [comment]
@zhuchun thanks. I totally missed `IViewController` part , looks very close to what I need. Will try it out and return with results

**Comment by c0ffeeartc** [comment]
I wrote TransformController and it solved a lot of issues, but I didn't have time for very thorough check. Here are my thoughts:
  - it's about immediate vs postponed state changes. Controller, Group Callbacks are immediate, Systems are postponed.
  - having immediate changes in transform is crucial when dealing with parented transforms. Immediately changing unity transforms removes performance gains, but is required for states heavily dependent on parenting. It's not that bad, just the regular way unity works.

Some implementation details:
  - I used controller for read and write, but kept positionLocal, rotationLocal, scaleLocal components for reacting
  - if entity has view, position, rotation or scale it must have a transformController. Each entity with TrController must have a gameobject
  - gameobject is added immediately

Here is gist to some parts [link](https://gist.github.com/c0ffeeartc/bfe63a98d0f2920a1a38423345c4c6ae)
